### PR TITLE
zkEVM update: Adding tx status check doc

### PR DESCRIPTION
**Pull request by EmpieichO**
zkEVM update: Adding tx status check doc ... as per ticket https://polygon.atlassian.net/browse/DOCS-266

### CODE REVIEW

File: docs/zkEVM/how-to/check-tx-status.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Here's how to verify the status of a transaction when using an RPC node as an intermediary step to the Polygon zkEVM network.
+
+This guide is for users who send transactions to an RPC node, which in turn relays these transactions to the Polygon zkEVM network.
+
+## Recommended endpoints:
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/check-tx-status.md]:
```suggestion
## Recommended endpoints
```

File: docs/zkEVM/how-to/check-tx-status.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+After sending a transaction (TX) to the network using the `eth_sendRawTransaction`, use the following endpoints to check the TX Status:
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/check-tx-status.md]:
```suggestion
After sending a transaction (TX) to the network using the `eth_sendRawTransaction`, use the following endpoints to check the status:
```

File: docs/zkEVM/how-to/check-tx-status.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+1. `eth_getTransactionByHash`
+2. `eth_getTransactionReceipt`
+
+### Using eth_getTransactionByHash
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/check-tx-status.md]:
```suggestion
### Using `eth_getTransactionByHash`
```

File: docs/zkEVM/how-to/check-tx-status.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+(a) If the result is null, it means either the TX doesn't exist in the network or it was discarded.
+
+(b) If the result contains some data, then
+
+- If the fields `block_num` and `block_hash` are null, it means the TX is still in the pool and is pending.
+
+- If the fields `block_num` and `block_hash` are NOT null, it means the TX was mined.
+
+### Using eth_getTransactionReceipt
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/check-tx-status.md]:
```suggestion
### Using `eth_getTransactionReceipt`
```

File: mkdocs.yml [hunk 5]
```diff
@@ -93,6 +93,7 @@ nav:
         - Deploy a contract with Foundry: zkEVM/how-to/using-foundry.md
         - Deploy a contract with Hardhat: zkEVM/how-to/using-hardhat.md
         - Verify a contract: zkEVM/how-to/verify-contract.md
+        - Checking transaction status: zkEVM/how-to/check-tx-status.md
```
Comment (kmurphypolygon) [inline mkdocs.yml]:
```suggestion
        - Check transaction status: zkEVM/how-to/check-tx-status.md
```

File: docs/zkEVM/how-to/check-tx-status.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+When checking the TX status using the `eth_getTransactionByHash`, the result can be either one of the following.
+
+(a) If the result is null, it means either the TX doesn't exist in the network or it was discarded.
+
+(b) If the result contains some data, then
+
+- If the fields `block_num` and `block_hash` are null, it means the TX is still in the pool and is pending.
+
+- If the fields `block_num` and `block_hash` are NOT null, it means the TX was mined.
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/check-tx-status.md:22]:
```suggestion
        - If the fields `block_num` and `block_hash` are null, it means the TX is still in the pool and is pending.

        - If the fields `block_num` and `block_hash` are NOT null, it means the TX was mined.
```

Comment (kmurphypolygon) [reply docs/zkEVM/how-to/check-tx-status.md:22]: can you tab these over so they are not high level bullets with a and b? I hacked it so it probably doesn't work.

Comment (EmpieichO) [reply docs/zkEVM/how-to/check-tx-status.md:22]: A tab turns them into a copiable-code form which ... causes text to be in one long line that needs to be scrolled to be viewed ...  not so desirable.

Comment (kmurphypolygon) [reply docs/zkEVM/how-to/check-tx-status.md:22]: If you change them to this it works: 

```
- If the result is null, it means either the TX doesn't exist or it is still pending to be mined. In this case, use the `eth_getTransactionByHash` endpoint to check it.

- If the result contains some data, it means the TX was already mined. However,

    a) If the field status is 1 (success), it means the TX affected the state as expected.

    b) If the field status is 0 (failure), it means the TX has consumed gas used while processing the TX, but hasn't changed the state as expected.
```

**Review by kmurphypolygon — APPROVED** [review]

**Pull request merged**